addons/crm/models/ir_http.py

```diff
@@ -0,0 +1,51 @@
+import base64
+from odoo import models
+from odoo.http import request
+import odoo.http
+from werkzeug import exceptions
+from hashlib import sha256
+import json
+import hmac
+import time
+
+class IrHttp(models.AbstractModel):
+    _inherit = 'ir.http'
+
+    @classmethod
+    def _auth_method_crmaddon(cls):
+        if request.httprequest.method == "OPTIONS":
+            return True
+
+########################
+        #raise exceptions.BadRequest("invalid token")
+        #raise odoo.http.SessionExpiredException("Invalid token")
+        #raise exceptions.Unauthorized('Invalid token')
+###########################
+        token = dict(request.httprequest.headers).get('Authorization')
```

Comment:
If you're going to use standard headers, please use them correctly: the Authorization header is <type> <credentials>.